grafana_wtf/core.py

```diff
@@ -223,7 +223,7 @@ def scan_dashboards(self, dashboard_uids=None):
         if self.progressbar:
             self.start_progressbar(len(self.data.dashboard_list))
 
-        if self.concurrency is None or self.concurrency <= 1:
+        if self.concurrency is None or self.concurrency <= 1 or hasattr(self.grafana.client.s, "cache"):
             self.fetch_dashboards()
         else:
             self.fetch_dashboards_parallel()
```

Comment:
Dear @glitchwizard,

apologies that using caching together with parallel fetching causes issues for you, and thanks for your patch.

https://github.com/grafana-toolbox/grafana-wtf/blob/25d5df50e0d16532151e7f8ed2f2b7fd36a0d60f/grafana_wtf/commands.py#L237-L238

I wonder if your change would effectively **always** turn off parallelism, because the program is apparently configured to enable both caching and concurrent/parallel fetching by default, which means `hasattr(self.grafana.client.s, "cache")` will always be true?

With kind regards,
Andreas.

Reply:
As the program is using the `niquests` library through `grafana-client`,
```python
class CachedSession(requests_cache.session.CacheMixin, niquests.Session)
```

I wonder if `requests_cache` and `niquests` do not play well together here, and what we could do about it. Is it a concern of using the packages together correctly, and how, or could either of them be improved to achieve the same?

We were already having a conversation started by @JensRichnow, who reported problems observed on an Apple M2 Max.

- https://github.com/grafana-toolbox/grafana-wtf/issues/111

Can you possibly also share which OS, Python version, and `grafana-wtf` version you are using?

/cc @Ousret, @JWCook

Reply:
> if your change would effectively **always** turn off parallelism

Of course, it will be applied and released if we don't find any better means of resolving the problem. In this case, I'd suggest to turn of concurrency by default, and only enable it when the user requests it, i.e. similar like your patch, but slightly different.

Reply:
@amotl honestly, I don't actually know enough about the issue to be able to solve in a way that would be better than I've proposed. 

Mostly because I'd have to teach myself about how it's supposed to work, and all that. I just needed a quick fix, so this PR could be canceled if it's not an effective enough fix, I don't know that I'd have time to add the flag to it and finish the PR in the manner you're requesting, though that does sound like a really great idea.

Reply:
Hi @glitchwizard. I've provided an alternative per GH-181. Can you verify it works for you, by following the guidelines within the "Preview" section of the PR description?

Reply:
I've just merged GH-181, and will cut a release right away. Let's continue to discuss in GH-185?

Reply:
For my part, I think that it is unlikely that Niquests caused the seg fault. I made sure that Niquests works under the freethreaded build. On the other hand, there is multiple reports of threading issue with requests_cache.
Unless we can have a minimal reproducer, I'd say Niquests is safe.